### PR TITLE
Add e2e tests for cinder CSI

### LINKED ISSUE #690: Improve functional testing (e2e) for Cinder CSI
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line: 
>
> /kind bug
> /kind feature

**What happened**:
No functional testing happening on gate for various features like topology, snapshot, block volume etc. We need to add the same

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- openstack-cloud-controller-manager version:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by ramineni**
**What this PR does / why we need it**:
This PR adds e2e test framework for cinder CSI driver.
This PR enables only dynamic provisioning tests.
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #690 

**Special notes for your reviewer**:
PR to enable the job in openlab is here https://github.com/theopenlab/openlab-zuul-jobs/pull/690
**Release note**:

```release-note
Adds e2e test framework for cinder csi driver.
```

### CODE REVIEW

File: cmd/tests/cinder_csi_e2e_suite_test.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	"log"
+	"os"
+	"path"
+	"testing"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo/config"
+	"github.com/onsi/ginkgo/reporters"
+	"github.com/onsi/gomega"
+	_ "k8s.io/cloud-provider-openstack/test"
```
Comment (adisky) [inline cmd/tests/cinder_csi_e2e_suite_test.go]: is this required?

File: test/csi-volumes.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package test
```
Comment (adisky) [inline test/csi-volumes.go]: any reasons for csi-volumes.go instead of csi_volumes.go?

File: cmd/tests/cinder_csi_e2e_suite_test.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
+package main
```
Comment (adisky) [inline cmd/tests/cinder_csi_e2e_suite_test.go]: any special reason for adding this inside cmd?
could we have pkg structure like this
/test/e2e/..
/test/e2e/cmd

Comment (ramineni) [reply cmd/tests/cinder_csi_e2e_suite_test.go]: The structure is a per in documentation https://kubernetes-csi.github.io/docs/functional-testing.html

File: cmd/tests/cinder_csi_e2e_suite_test.go (hunk 1 above)
Comment (ramineni) [reply cmd/tests/cinder_csi_e2e_suite_test.go]: yes , its required to load, else gives error

File: test/csi-volumes.go (hunk 2 above)
Comment (ramineni) [reply test/csi-volumes.go]: its as per documentation , it is mentioned to use that, https://kubernetes-csi.github.io/docs/functional-testing.html

**Comment by adisky** [comment]
Thanks!!
/lgtm

File: .zuul.yaml [hunk 4]
```diff
@@ -64,6 +64,15 @@
...
               - ^.gitignore$
+    cloud-provider-openstack-e2e-test-csi-cinder:
+      jobs:
+        - cloud-provider-openstack-e2e-test-csi-cinder:
+            irrelevant-files:
+              - ^docs/.*$
+              - ^.*\.md$
+              - ^OWNERS$
+              - ^SECURITY_CONTACTS$
+              - ^.gitignore$
```
Comment (lingxiankong) [inline .zuul.yaml]: Is it possible to run e2e test when only the cinder csi related code is changed?

Comment (ramineni) [reply .zuul.yaml]: done

**Comment by lingxiankong** [comment]
/lgtm

Thanks!

**Pull request merged**